### ISSUE TITLE
Reports will not be emailed when using Google Site Kit

### ISSUE THREAD

**Issue by apmwebdev** [woocommerce/woocommerce-admin#4778]
**Describe the bug**
WooCommerce Admin appears to be incompatible with Google Site Kit, at least as far as emailing reports. The emails with the report fail to send due to the scheduled action failing. I haven't installed GSK on a site to test this myself, but I've seen multiple customer sites where this is an issue. Given the ubiquity of Google Site Kit and the number of stores that will have reports emailed instead of downloaded immediately, this is a compatibility issue that will likely affect a lot of merchants.

Looking at the scheduled actions, you see this:

```
woocommerce_admin_report_export Failed
0 => 5
1 => ‘15928136251358’
2 => ‘customers’
3 => array ( ‘orderby’ => ‘date_last_active’, ‘order’ => ‘desc’, ‘page’ => 1, ‘per_page’ => 25, )
wc-admin-data Non-repeating 2020-06-22 08:13:50 +0000
(23 hours 25 minutes ago)
2020-06-22 08:13:45 +0000
action created
2020-06-22 08:17:43 +0000
action started via WP Cron
2020-06-22 08:17:43 +0000
unexpected shutdown: PHP Fatal error Uncaught TypeError: Argument 1 passed to Google\Site_Kit\Core\REST_API\REST_Routes::register_routes() must be an instance of WP_REST_Server, string given, called in /home/customer/www/firstsmile.pk/public_html/wp-content/plugins/google-site-kit/includes/Core/REST_API/REST_Routes.php on line 93 and defined in /home/customer/www/firstsmile.pk/public_html/wp-content/plugins/google-site-kit/includes/Core/REST_API/REST_Routes.php:105 Stack trace: #0 /home/customer/www/firstsmile.pk/public_html/wp-content/plugins/google-site-kit/includes/Core/REST_API/REST_Routes.php(93): Google\Site_Kit\Core\REST_API\REST_Routes->register_routes(”) #1 /home/customer/www/firstsmile.pk/public_html/wp-includes/class-wp-hook.php(287): Google\Site_Kit\Core\REST_API\REST_Routes->Google\Site_Kit\Core\REST_API\{closure}(”) #2 /home/customer/www/firstsmile.pk/public_html/wp-includes/class-wp-hook.php(311): WP_Hook->apply_filters(NULL, Array) #3 /home/customer/www/firstsmile.pk/public_html/wp-includes/plugin.php(478): WP_Hook->do_action(A in /home/customer/www/firstsmile.pk/public_html/wp-content/plugins/google-site-kit/includes/Core/REST_API/REST_Routes.php on line 105
```

**To Reproduce**
Try to export a report that needs to be emailed when GSK is installed.

**Expected behavior**
Reports should email correctly even when Google Site Kit is installed. I realize that compatibility issues aren't always things we can easily fix, but Google Site Kit is fairly common, so I thought it was worth creating an issue for just to create awareness at the very least.

**Additional context**
I can't find the previous tickets/forum threads about this, but the most recent one is [here](https://wordpress.org/support/topic/your-customer-report-has-been-sent-to-your-email-issue/). The SSR for that site is:

<details>

```
### WordPress Environment ###

WordPress address (URL): https://www.firstsmile.pk
Site address (URL): https://www.firstsmile.pk
WC Version: 4.0.1
REST API Version: ✔ 1.0.7
WC Blocks Version: ✔ 2.5.14
Action Scheduler Version: ✔ 3.1.4
WC Admin Version: ✔ 1.0.3
Log Directory Writable: ✔
WP Version: ❌ 5.4 – There is a newer version of WordPress available (5.4.2)
WP Multisite: –
WP Memory Limit: 768 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: Apache
PHP Version: 7.3.18
PHP Post Max Size: 128 MB
PHP Time Limit: 120
PHP Max Input Vars: 3000
cURL Version: 7.66.0
OpenSSL/1.1.1g-fips

SUHOSIN Installed: –
MySQL Version: 5.7.25-28-log
Max Upload Size: 128 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 4.0.1
WC Database Prefix: wp_
Total Database Size: 23.61MB
Database Data Size: 18.53MB
Database Index Size: 5.08MB
wp_woocommerce_sessions: Data: 1.91MB + Index: 0.06MB + Engine MyISAM
wp_woocommerce_api_keys: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_woocommerce_attribute_taxonomies: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_woocommerce_downloadable_product_permissions: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_woocommerce_order_items: Data: 0.04MB + Index: 0.02MB + Engine MyISAM
wp_woocommerce_order_itemmeta: Data: 0.32MB + Index: 0.25MB + Engine MyISAM
wp_woocommerce_tax_rates: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_woocommerce_tax_rate_locations: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_woocommerce_shipping_zones: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_woocommerce_shipping_zone_locations: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_woocommerce_shipping_zone_methods: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_woocommerce_payment_tokens: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_woocommerce_payment_tokenmeta: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_woocommerce_log: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_actionscheduler_actions: Data: 1.02MB + Index: 0.44MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 1.52MB + Index: 0.42MB + Engine InnoDB
wp_captured_wc_fields: Data: 0.06MB + Index: 0.00MB + Engine InnoDB
wp_cartflows_ca_cart_abandonment: Data: 0.08MB + Index: 0.02MB + Engine InnoDB
wp_cartflows_ca_email_history: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_cartflows_ca_email_templates: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_cartflows_ca_email_templates_meta: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_commentmeta: Data: 0.00MB + Index: 0.01MB + Engine MyISAM
wp_comments: Data: 0.53MB + Index: 0.35MB + Engine MyISAM
wp_failed_jobs: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_imagify_files: Data: 0.02MB + Index: 0.08MB + Engine InnoDB
wp_imagify_folders: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_links: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_mailchimp_carts: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_mailchimp_jobs: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_options: Data: 3.79MB + Index: 0.70MB + Engine MyISAM
wp_postmeta: Data: 7.02MB + Index: 1.51MB + Engine MyISAM
wp_posts: Data: 0.79MB + Index: 0.35MB + Engine MyISAM
wp_queue: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_rank_math_404_logs: Data: 0.01MB + Index: 0.02MB + Engine MyISAM
wp_rank_math_internal_links: Data: 0.03MB + Index: 0.01MB + Engine MyISAM
wp_rank_math_internal_meta: Data: 0.05MB + Index: 0.04MB + Engine MyISAM
wp_rank_math_redirections: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_rank_math_redirections_cache: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_rank_math_sc_analytics: Data: 0.37MB + Index: 0.16MB + Engine MyISAM
wp_revslider_css: Data: 0.09MB + Index: 0.00MB + Engine MyISAM
wp_revslider_css_bkp: Data: 0.09MB + Index: 0.00MB + Engine MyISAM
wp_revslider_layer_animations: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_revslider_layer_animations_bkp: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_revslider_navigations: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_revslider_navigations_bkp: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_revslider_sliders: Data: 0.01MB + Index: 0.00MB + Engine MyISAM
wp_revslider_sliders_bkp: Data: 0.01MB + Index: 0.00MB + Engine MyISAM
wp_revslider_slides: Data: 0.05MB + Index: 0.00MB + Engine MyISAM
wp_revslider_slides_bkp: Data: 0.10MB + Index: 0.00MB + Engine MyISAM
wp_revslider_static_slides: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_revslider_static_slides_bkp: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_smush_dir_images: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_soo_wishlists: Data: 0.00MB + Index: 0.01MB + Engine MyISAM
wp_soo_wishlists_items: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_termmeta: Data: 0.05MB + Index: 0.03MB + Engine MyISAM
wp_terms: Data: 0.01MB + Index: 0.02MB + Engine MyISAM
wp_term_relationships: Data: 0.09MB + Index: 0.18MB + Engine MyISAM
wp_term_taxonomy: Data: 0.01MB + Index: 0.01MB + Engine MyISAM
wp_usermeta: Data: 0.09MB + Index: 0.06MB + Engine MyISAM
wp_users: Data: 0.00MB + Index: 0.01MB + Engine MyISAM
wp_wc_admin_notes: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_wc_admin_note_actions: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_wc_category_lookup: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.02MB + Engine MyISAM
wp_wc_download_log: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_wc_order_coupon_lookup: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_wc_order_product_lookup: Data: 0.05MB + Index: 0.05MB + Engine MyISAM
wp_wc_order_stats: Data: 0.02MB + Index: 0.02MB + Engine MyISAM
wp_wc_order_tax_lookup: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_wc_product_meta_lookup: Data: 0.06MB + Index: 0.07MB + Engine MyISAM
wp_wc_tax_rate_classes: Data: 0.00MB + Index: 0.01MB + Engine MyISAM
wp_wc_webhooks: Data: 0.00MB + Index: 0.00MB + Engine MyISAM

### Post Type Counts ###

acf-field: 1
acf-field-group: 1
attachment: 642
ct_size_guide: 128
custom_css: 1
customize_changeset: 1
mc4wp-form: 1
nav_menu_item: 36
page: 20
post: 3
product: 179
product_variation: 502
revision: 40
shop_coupon: 51
shop_order: 227
shop_order_refund: 1
tabs_responsive: 1
wpcf7_contact_form: 2

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (30) ###

Advanced Custom Fields: by Elliot Condon – 5.8.9
Classic Editor: by WordPress Contributors – 1.5
Contact Form 7: by Takayuki Miyoshi – 5.1.7
createIT Size Guide Plugin: by createIT – 2.6 – Not tested with the active version of WooCommerce
Disable All WordPress Updates: by Oliver Schlöbe – 1.6.6
Duplicate Post: by Enrico Battocchi – 3.2.4
Site Kit by Google: by Google – 1.7.0
Imagify: by WP Media – 1.9.9
WPBakery Page Builder: by Michael M – WPBakery.com – 6.1
Kirki Customizer Framework: by Ari Stathopoulos (@aristath) – 3.1.1
Mailchimp for WooCommerce: by Mailchimp – 2.3.6 – Not tested with the active version of WooCommerce
MC4WP: Mailchimp for WordPress: by ibericode – 4.7.6
Meta Box: by MetaBox.io – 5.2.10
Official Facebook Pixel: by Facebook – 2.0.1
Perfect WooCommerce Brands: by QuadLayers – 1.8.3 – Not tested with the active version of WooCommerce
RY WC City Select: by Richer Yang – 1.0.9 – Not tested with the active version of WooCommerce
Rank Math SEO: by Rank Math – 1.0.42.1
Sober Addons: by UIX Themes – 1.4.1
Soo Product Filter: by UIX Themes – 1.0.8 – Not tested with the active version of WooCommerce
Soo Wishlist: by UIX Themes – 1.1.2 – Not tested with the active version of WooCommerce
Top Bar: by WP Darko – 2.0
Variation Swatcher for WooCommerce: by ThemeAlien – 1.0.7 – Not tested with the active version of WooCommerce
WooCommerce Cart Abandonment Recovery: by CartFlows Inc – 1.2.5 – Not tested with the active version of WooCommerce
CartBounty – Save and recover abandoned carts for WooCommerce: by Streamline.lv – 4.4.1 – Not tested with the active version of WooCommerce
Show Single Variations Shop & Category for WooCommerce: by Gravity Master – 2.0 – Not tested with the active version of WooCommerce
WooCommerce Admin: by WooCommerce – 1.0.3 – Not tested with the active version of WooCommerce
WooCommerce Checkout Field Editor: by WooCommerce – 1.5.30 – Not tested with the active version of WooCommerce
WooCommerce Custom Product Tabs Lite: by SkyVerge – 1.7.3 – Not tested with the active version of WooCommerce
WooCommerce Google Analytics Integration: by WooCommerce – 1.4.20 – Not tested with the active version of WooCommerce
WooCommerce: by Automattic – 4.0.1

### Inactive Plugins (4) ###

Google Tag Manager for WordPress: by Thomas Geiger – 1.11.4
Hummingbird: by WPMU DEV – 2.4.2
W3 Total Cache: by BoldGrid – 0.13.2
WP Rocket: by WP Media – 3.5.0.2

### Settings ###

API Enabled: –
Force SSL: –
Currency: PKR (PKR)
Currency Position: left_space
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 0
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: ✔

### WC Pages ###

Shop base: #7 – /shop/
Cart: #8 – /cart/
Checkout: #9 – /checkout/
My account: #10 – /my-account/
Terms and conditions: #3 – /privacy-policy/

### Theme ###

Name: Sober Child Theme
Version: 1.0.0
Author URL: http://uix.store
Child Theme: ✔
Parent Theme Name: Sober
Parent Theme Version: 2.3.0
Parent Theme Author URL: http://uix.store
WooCommerce Support: ✔

### Templates ###

Overrides: sober/woocommerce/cart/cart-shipping.php
sober-child/woocommerce/cart/cart.php
sober/woocommerce/cart/mini-cart.php
sober-child/woocommerce/checkout/form-checkout.php
sober/woocommerce/checkout/form-coupon.php
sober/woocommerce/checkout/form-login.php
sober/woocommerce/checkout/review-order.php
sober/woocommerce/global/form-login.php
sober/woocommerce/global/quantity-input.php
sober/woocommerce/global/wrapper-end.php
sober/woocommerce/global/wrapper-start.php
sober/woocommerce/loop/add-to-cart.php
sober/woocommerce/loop/price.php
sober/woocommerce/loop/result-count.php
sober/woocommerce/myaccount/dashboard.php
sober/woocommerce/myaccount/form-edit-account.php
sober/woocommerce/myaccount/form-login.php
sober/woocommerce/myaccount/my-address.php
sober/woocommerce/notices/error.php
sober/woocommerce/notices/success.php
sober/woocommerce/order/form-tracking.php
sober/woocommerce/single-product/add-to-cart/external.php
sober/woocommerce/single-product/add-to-cart/grouped.php
sober/woocommerce/single-product/add-to-cart/simple.php
sober/woocommerce/single-product/add-to-cart/variable.php
sober/woocommerce/single-product/add-to-cart/variation-add-to-cart-button.php
sober/woocommerce/single-product/product-image.php
sober/woocommerce/single-product/product-thumbnails.php
sober/woocommerce/single-product/related.php
sober/woocommerce/single-product/review-meta.php
sober/woocommerce/single-product/tabs/additional-information.php
sober-child/woocommerce/single-product/tabs/tabs.php
sober/woocommerce/single-product/up-sells.php
sober/woocommerce/single-product-reviews.php

### Action Scheduler ###

Canceled: 1
Oldest: 2020-06-12 04:28:17 +0000
Newest: 2020-06-12 04:28:17 +0000

Complete: 1,370
Oldest: 2020-05-22 09:25:14 +0000
Newest: 2020-06-22 08:37:10 +0000

Failed: 100
Oldest: 2020-01-29 14:34:28 +0000
Newest: 2020-06-22 08:17:43 +0000

Pending: 2
Oldest: 2020-06-22 09:26:03 +0000
Newest: 2020-06-23 07:06:01 +0000
```
</details>

**Comment by timmyc** [comment]
@apmwebdev a fix for this was shipped in 1.3.0/4.3.0

https://github.com/woocommerce/woocommerce-admin/pull/4691

So if any further reports are happening, the solution is to upgrade WooCommerce Admin or WooCommerce. I'll close this out as already fixed, but please do re-open if the problem persists.

**Comment by apmwebdev** [comment]
Oh, OK, that makes sense. Thanks @timmyc!
